console/console-init/ui/src/modules/project-detail/components/ProjectCertificateList/CertificateForm.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState, useEffect } from "react";
+import {
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  TextInput,
+  Grid,
+  GridItem,
+  Split,
+  SplitItem,
+  ActionGroup,
+  Button,
+  Switch,
+  DropdownPosition
+} from "@patternfly/react-core";
+import { DropdownWithToggle } from "components";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { getCertificateLabelByKey } from "modules/project-detail/utils/utils";
+import { algorithmTypeOptions } from "modules/project-detail/utils/constant";
+import { IProjectCertificate } from "./ProjectCertificateList";
+
+const styles = StyleSheet.create({
+  dropdown_align: { display: "flex" },
+  dropdown_toggle_align: { flex: "1" }
+});
+
+export interface ICertificateFormProps {
+  certificate?: IProjectCertificate;
+  setOnEditMode: React.Dispatch<React.SetStateAction<boolean>>;
+  id: string;
+}
+
+export const CertificateForm: React.FunctionComponent<ICertificateFormProps> = ({
+  setOnEditMode,
+  certificate,
+  id
+}) => {
+  const [subjectDn, setSubjectDn] = useState<string>();
+  const [publicKey, setPublicKey] = useState<string>();
+  const [algorithm, setAlgorithm] = useState<string>();
+  const [notBefore, setNotBefore] = useState<string>();
+  const [notAfter, setNotAfter] = useState<string>();
+  const [autoProvision, setAutoProvision] = useState<boolean>();
+
+  useEffect(() => {
+    // code to initialize the form if it is being used for edit
+    if (certificate) {
```

Comment:
Not sure, why do you need useEffect and if things are depends on certificate, you must pass certificate as dependancy.

Reply:
There are two use cases of this form - one is to edit an existing certificate and another to create a new certificate. The second case doesn't need a prop.